internal/services/nginx/nginx_deployment_data_source.go

```diff
@@ -267,7 +291,19 @@ func (m DeploymentDataSource) Read() sdk.ResourceFunc {
 					}
 
 					if scaling := props.ScalingProperties; scaling != nil {
```

Comment:
UX question, do we want to check and send a error message here for only allowing manual or autoscale or are we going to rely on the API 400 request?

Reply:
not relevant in this PR anymore. Going to resolve it.

Also, we should rely on the API for quite a lot but we can (will have to) handle these conflicting cases in Terraform.